dodiscover/ci/fisher_z_test.py

```diff
@@ -0,0 +1,107 @@
+from math import log, sqrt
+from typing import Any, Set, Tuple, Union
+
+import numpy as np
+import pandas as pd
+from scipy.stats import norm
+
+from .base import BaseConditionalIndependenceTest
+
+
+class FisherZCITest(BaseConditionalIndependenceTest):
+    def __init__(self, correlation_matrix=None):
+        """Conditional independence test using Fisher-Z's test for Gaussian random variables.
+
+        Parameters
+        ----------
+        correlation_matrix : np.ndarray of shape (n_variables, n_variables), optional
+            ``None`` means without the parameter of correlation matrix and
+            the correlation will be computed from the data., by default None
+        """
+        self.correlation_matrix = correlation_matrix
+
+    def test(
+        self, df: pd.DataFrame, x_var: Any, y_var: Any, z_covariates: Any = None
+    ) -> Tuple[float, float]:
```

Comment:
Might be useful to create types for the test stat and the p-value as this will be used in multiple places.

Reply:
Aren't they just floats tho? With pvalue being constrained between 0 and 1